cmd/factory_test.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+func TestEnvPathFactory(t *testing.T) {
+	path := "fake_path"
+	home, err := homedir.Dir()
+	if err != nil {
+		t.Errorf("Failed to find the home directory: %s", err.Error())
+	}
+
+	tests := []struct {
+		setQRI     bool
+		setIPFS    bool
+		path       string
+		qriAnswer  string
+		ipfsAnswer string
+	}{
+		{false, false, path, filepath.Join(home, ".qri"), filepath.Join(home, ".ipfs")},
+		{true, false, path, path, filepath.Join(home, ".ipfs")},
+		{false, true, path, path, path},
+	}
+
+	//Note: Variables created with os.Setenv exist for the rest of the process
```

Comment:
## defer
Using defer to clean up after myself makes sense. Will do.

## the note
The defer makes sense, but I think it is unrelated to the note because the note is about the state change.

os.Getenv(key string) returns the value of the key, but, if the key does not exist, it returns an empty string.

On my machine, the environment variables "QRI_PATH" and "IPFS_PATH" do not exist. Once I set them with os.Setenv, they then existed for the rest of the running process. The defer will change the values of these environment variables to their original state, which, in my case, in an empty string, but the created environment variables will still exist, which is a temporary change of state (because they did not exist before this process) that could affect other tests (or test cases).

I can elaborate the note so that it is clearer?

## os.Getwd
I am not sure why you would want to change, or keep track of, the working directory. As far as I can tell, the EnvPathFactory function does not change or modify the working directory. Thus, adding those lines should not affect the test, but I do not understand why it is needed. Can you explain?

Reply:
In regard to the `os.Getwd`, oh my I'm sorry. This bit is coming from a long-since removed code path that used to be working-directory dependent. You're absolutely right that we don't need to do anything that changes the working directory for these tests. Apologies for the mixup